### PR TITLE
fix big letters in [(sub)|(super)]script

### PR DESCRIPTION
**Pull request by MeggyCal**
When we have big roman letters in subscript, it looks uggly (for example $a\_D$). So this macro makes them smaller.

### CODE REVIEW

File: texmf/tex/latex/fks/fkssugar.sty [hunk 1]
```diff
@@ -27,8 +27,34 @@
...
+\setbox\my=\hbox{\rm Á}
+\bigletterdashed=\ht\my		%so do dashed big letters (in sp, e.g. number)
+\setbox\my=\hbox{$\rm #1$}
+\sz=\ht\my 			%convert height to scaled point, e.g. number
+}
+
+\renewcommand\_[1]{\ifmmode
+\mmbox{#1}
+\ifnum\sz=\bigletter _{\rm\scriptscriptstyle #1}
+\else \ifnum\sz=\bigletterdashed _{\rm \scriptscriptstyle #1}
```
[the remark this answers is not in the document]
Comment (LukasLedvina) [reply texmf/tex/latex/fks/fkssugar.sty]: Are you sure that `\ht` of each text containing at least one capital letter is `bigletterdashed` or `bigletter`??. I'd prefer sth. like (in pseudocode)  `\ht > \smallletter + epsilon`. Also the shouled be written tests containg letters like "j", "i", "Ů" (\ht od Ů might differ from  \ht of Á), also `$x\^{\dot a}$` `$x\^{\dot A}$` (math accents), numbers,...

In general how about making this as configurable feature??

**Comment by MeggyCal** [comment]
@LukasLedvina \textnormal leaves math mode, so you can't use for example \_\Gamma. I believe, anyway, that \rm behaves like a switch which sets main font of the document (not always latin), although I tested it only once. @Werkov I don't know how it is with beamer, maybe it doesn't have correctly set main font? By the way, I don't like that sans serif font anyway.

I see using \textnormal would be better for beamer, but it disables math symbols. We can scale the text in both cases with \scalebox.

File: texmf/tex/latex/fks/fkssugar.sty (hunk 1 above)
[the remark this answers is not in the document]
Comment (MeggyCal) [reply texmf/tex/latex/fks/fkssugar.sty]: Yep... and I have to admit that I am not sure about the same heigh. I tested it in plainTeX and XeLaTeX, I measured the letters in plainTeX. Today I measured them in XeLaTeX and the heights are different... I am really sorry, I thought they both use Latin Modern.

Making this as configurable feature is the only option... which means that I can hapilly delete everything I've already done on this macro (again). We still have to decide whether we want to enable math characters.

**Comment by MeggyCal** [comment]
@LukasLedvina @Werkov @timkol Considering the bug that in XeLaTeX default font are the heights of big letters different I agree with Lukáš L. to make the size of subscript as a configurable feature. It can be done by a fairly minor change:

\renewcommand\_[2][1]{\ifmmode _{\textnormal{\scalebox{#1}{$#2$}}}\else\underscore#2\fi} %roman subscript

but it still disables math symbols. Is it a problem for us?

**Comment by LukasLedvina** [comment]
The new pull request looks much better for me. May be the default scaling should be introduced to be consistent in documents. Like `\def\fkssmallscript{0.8}`.

I think that we would not need to use math in roman super/subscripts. The main manner for the roman sub/superscripts is shortening longer text expression. If someone want to to use it could be typeset by hand.

About `\rm` and `\textnormal` you can read for example [here](https://www.dickimaw-books.com/latex/novices/html/fontstyle.html). Beamer font configuration is +- correct, but be must not use `\rm`, `\bf`...

**Pull request merged**